### ISSUE TITLE
Vhost based administration rights

### ISSUE THREAD

**Issue by hairyhum** [rabbitmq/rabbitmq-server#682]
Related to https://github.com/rabbitmq/rabbitmq-management/issues/153
There should be some way to define admin rights for specific vhost.

**Comment by michaelklishin** [comment]
I suggest that we put this off until we begin working on topic-based authorisation. While this is not directly related, such things need to be designed as a group.

**Comment by megwill4268** [comment]
Hi @michaelklishin , excited to have the topic-based authorizations implemented. This feature would be extraordinarily useful to my organization (and personally I could stop maintaining an application to hack around this) is this still in sight to implement?

**Comment by lukebakken** [comment]
@megwill4268 RabbitMQ implements topic authorization as of 3.7.0. Read about it [here](https://www.rabbitmq.com/access-control.html#topic-authorisation).

**Comment by megwill4268** [comment]
@michaelklishin yes, I was thanking you for the topic authorization feature. I was asking if you were still looking at implementing vhost level administration of users and permissions. https://github.com/rabbitmq/rabbitmq-management/issues/153

**Comment by michaelklishin** [comment]
User permissions are per-vhost and have always been. The issue description is very vague but I suspect that it's a about having a "vhost-specific administrator" since currently administrators can manage everything across all vhosts. This is not on the roadmap for the next release.

**Comment by Skeen** [comment]
While I cannot give a suggestion for what a solution would look like, I can suggest how I would like the behavior to be.

I would like to have an instance admin, which creates namespaces (read vhosts) and create a namespace admin (vhost admin).
Namespace admin should be able to control all resources within their own namespace, but no resources outside of it.
The namespace admin should be able to create namespaced users, namespaces exchanged, queues, etc.

I believe this proper namespace isolation is necessary to support multi-tenancy in RabbitMQ, as currently the RBAC is not strong enough to provide proper separation of tenants.
